### PR TITLE
Navbar

### PR DESCRIPTION
**Pull request by brunnermanu**
Add navbar, home and about-me-component

screely-1672841019321

### CODE REVIEW

File: src/app/button/button.component.scss [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+  padding: 5px 15px;
+  outline: none;
+  transition: 0.2s ease-in-out;
+  text-align: center;
+  @include headings(button-text)
+}
+
+.button:hover {
+  cursor: pointer;
+}
```
Comment (khanlifts) [inline src/app/button/button.component.scss]:
```suggestion
  @include headings(button-text)
  &:hover {
    cursor: pointer;
  }
```

File: src/app/app-routing.module.ts [hunk 2]
```diff
@@ -1,10 +1,17 @@
 import { NgModule } from '@angular/core';
 import { RouterModule, Routes } from '@angular/router';
+import { AboutMeComponent } from "./about-me/about-me.component";
+import { HomeComponent } from "./home/home.component";
 
-const routes: Routes = [];
+const routes: Routes = [
+  {path: 'home', component: HomeComponent},
```
Comment (khanlifts) [inline src/app/app-routing.module.ts]:
```suggestion
  { path: 'home', component: HomeComponent },
```

File: src/app/app.component.html [hunk 3]
```diff
@@ -1,484 +1,4 @@
...
-</div>
-
-<!-- * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * -->
-<!-- * * * * * * * * * * * The content above * * * * * * * * * * * -->
-<!-- * * * * * * * * * * is only a placeholder * * * * * * * * * * -->
-<!-- * * * * * * * * * * and can be replaced. * * * * * * * * * * * -->
-<!-- * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * -->
-<!-- * * * * * * * * * * End of Placeholder * * * * * * * * * * * -->
-<!-- * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * -->
+<app-navbar></app-navbar>
```
Comment (khanlifts) [inline src/app/app.component.html:1]: empty first line

File: src/app/button/button.component.html [hunk 4]
```diff
@@ -0,0 +1,4 @@
+<button class="button"
+        type="button">
+  <img *ngIf="iconPath" src="{{iconPath}}">
+  {{buttonName}}</button>
```
Comment (khanlifts) [inline src/app/button/button.component.html]:
```suggestion
  {{buttonName}}
</button>
```

File: src/app/button/button.component.scss [hunk 5]
```diff
@@ -0,0 +1,24 @@
...
+
+;
```
Comment (khanlifts) [inline src/app/button/button.component.scss]: Can you remove this?

File: src/app/button/button.component.scss [hunk 6]
```diff
@@ -0,0 +1,24 @@
...
+  background-color: $blue;
+  min-width: 60px;
+  display: flex;
+  flex-wrap: nowrap;
+  align-items: center;
+  gap: 10px;
+  border-radius: 4px;
+  padding: 5px 15px;
+  outline: none;
+  transition: 0.2s ease-in-out;
```
Comment (khanlifts) [inline src/app/button/button.component.scss:14]: Can you help me see where the transition is needed?

File: src/app/button/button.component.ts [hunk 7]
```diff
@@ -0,0 +1,14 @@
...
+  selector: 'app-button',
+  templateUrl: './button.component.html',
+  styleUrls: ['./button.component.scss']
+})
+export class ButtonComponent {
+  @Input() buttonName: string | undefined;
+  @Input() iconPath: string | undefined;
+
+  constructor() {
+  }
```
Comment (khanlifts) [inline src/app/button/button.component.ts]: remove

File: src/app/home/home.component.ts [hunk 8]
```diff
@@ -0,0 +1,10 @@
+import { Component } from '@angular/core';
+
+@Component({
+  selector: 'app-home',
+  templateUrl: './home.component.html',
+  styleUrls: ['./home.component.scss']
+})
+export class HomeComponent {
+
```
Comment (khanlifts) [inline src/app/home/home.component.ts:9]: empty line

File: src/styles/variables.scss [hunk 9]
```diff
@@ -0,0 +1,77 @@
...
+$grey: #C0C0C0;
+$light-grey: #E4E5E9;
+$extra-light: #EFF0F3;
+$white: #FFFFFE;
+$blue: #6246EA;
+$border-dark: rgba($dark, 0.88);
+
+.alert {
+  border: 1px solid $border-dark;
+}
```
Comment (khanlifts) [inline src/styles/variables.scss]: Will you use this class often?
If not, can you set it directly in the component as it's only one line?

File: src/app/navbar/navbar.component.html [hunk 10]
```diff
@@ -0,0 +1,11 @@
+<nav class="navbar">
+  <div class="logo">
```
Comment (khanlifts) [inline src/app/navbar/navbar.component.html]: Would it make sense to also add a `routeLink="home"` here?

File: src/app/button/button.component.ts [hunk 11]
```diff
@@ -0,0 +1,14 @@
+import { Component, Input } from "@angular/core";
+
+@Component({
+  selector: 'app-button',
+  templateUrl: './button.component.html',
+  styleUrls: ['./button.component.scss']
+})
+export class ButtonComponent {
+  @Input() buttonName: string | undefined;
+  @Input() iconPath: string | undefined;
```
Comment (khanlifts) [inline src/app/button/button.component.ts]: Check if `undefined` needed.

File: src/app/button/button.component.scss (hunk 6 above)
Comment (brunnermanu) [reply src/app/button/button.component.scss:14]: for hover styles

File: src/app/home/home.component.ts (hunk 8 above)
Comment (khanlifts) [reply src/app/home/home.component.ts:9]: I accept that you ignored me here.

**Review by khanlifts — APPROVED** [review]

**Pull request merged**